assets/pages/new-contributions-landing/components/StripePaymentRequestButton.jsx

```diff
@@ -0,0 +1,198 @@
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import { logException } from 'helpers/logger';
+import type { State } from '../contributionsLandingReducer';
+import {
+  setCanMakeApplePayPayment,
+  setStripePaymentRequestButtonClicked,
+  setPaymentRequest,
+  onStripePaymentRequestApiPaymentAuthorised,
+  updateEmail,
+} from '../contributionsLandingActions';
+
+
+// ----- Types -----//
+
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {|
+  stripe: Object,
+  paymentRequest: Object | null,
+  canMakeApplePayPayment: boolean,
+  country: IsoCountry,
+  currency: IsoCurrency,
+  setCanMakeApplePayPayment: (boolean) => void,
+  setPaymentRequest: (Object) => void,
+  onPaymentAuthorised: (PaymentAuthorisation, Function) => void,
+  setStripePaymentRequestButtonClicked: () => void,
+  isTestUser: boolean,
+  updateEmail: string => void,
+  selectedAmounts: SelectedAmounts,
+  otherAmounts: OtherAmounts,
+  contributionType: ContributionType,
+  countryGroupId: CountryGroupId,
+  amount: number,
+|};
+
+
+const mapStateToProps = (state: State) => ({
+  selectedAmounts: state.page.form.selectedAmounts,
+  otherAmounts: state.page.form.formData.otherAmounts,
+  canMakeApplePayPayment: state.page.form.stripePaymentRequestButtonData.canMakeApplePayPayment,
+  paymentRequest: state.page.form.stripePaymentRequestButtonData.paymentRequest,
+  countryGroupId: state.common.internationalisation.countryGroupId,
+  country: state.common.internationalisation.countryId,
+  currency: state.common.internationalisation.currencyId.toLowerCase(),
+  isTestUser: state.page.user.isTestUser || false,
+  contributionType: state.page.form.contributionType,
+});
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  onPaymentAuthorised:
+    (token, complete) => { dispatch(onStripePaymentRequestApiPaymentAuthorised(token, complete)); },
+  setCanMakeApplePayPayment:
+    (canMakeApplePayPayment) => { dispatch(setCanMakeApplePayPayment(canMakeApplePayPayment)); },
+  setPaymentRequest:
+    (paymentRequest) => { dispatch(setPaymentRequest(paymentRequest)); },
+  updateEmail: (email: string) => { dispatch(updateEmail(email)); },
+  setStripePaymentRequestButtonClicked: () => { dispatch(setStripePaymentRequestButtonClicked()); },
+});
+
+
+// ----- Functions -----//
+
+
+function updateUserEmail(data: Object, setEmail: string => void) {
+  const email = data.payerEmail;
+  if (email) {
+    if (isValidEmail(email)) {
+      setEmail(email);
+    } else {
+      logException(`Failed to set email for stripe payment request user with email: ${email}`);
+    }
+  } else {
+    logException('Failed to set email: no email in data object');
+  }
+}
+
+const onComplete = (complete: Function) => (res: PaymentResult) => {
+  if (res.paymentStatus === 'success') {
+    complete('success');
+  } else if (res.paymentStatus === 'failure') {
+    complete('fail');
+  }
+};
+
+function onClick(event, props: PropTypes) {
```

Comment:
could you put a comment explaining why we need to intercept the click event ourselves?

Reply:
👍